### PR TITLE
[CPDLP-1254] Delivery partner role and associations added

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-1254

### Changes proposed in this pull request

* Created `DeliveryPartnerProfile` for users
* Added seed user `delivery-partner@example.com` for testing
* New welcome email for delivery partner users

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -743,9 +752,9 @@
     t.string "declaration_type", null: false
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
-    t.index ["milestone_id", "schedule_id", "declaration_type"], name: "milestones_schedules_schedule_milestone_declaration_type", unique: true
+    t.index ["milestone_id", "schedule_id"], name: "index_schedule_milestones_on_milestone_id_and_schedule_id"
```
Comment (asmega) [inline db/schema.rb:755]: if there's no associated migration then i think we should remove this

Comment (mooktakim) [reply db/schema.rb]: Not sure where that came from. dropped and recreated local db

File: app/policies/delivery_partner_profile_policy.rb [hunk 2]
```diff
@@ -1,17 +1,19 @@
 # frozen_string_literal: true
 
 class DeliveryPartnerProfilePolicy < ApplicationPolicy
-  def create?
```
Comment (StupidCodeFactory) [inline app/policies/delivery_partner_profile_policy.rb:4]: Weren't those used? could very well be that they weren't, Might be worth asking in the dev channel

File: app/views/delivery_partners/participants/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<h1 class="govuk-heading-xl">Participants</h1>
+
+<p>Hello world</p>
```
Comment (StupidCodeFactory) [inline app/views/delivery_partners/participants/index.html.erb:3]: famous last words :)

File: config/routes.rb [hunk 4]
```diff
@@ -498,6 +498,11 @@
     end
   end
 
+  namespace :delivery_partners, path: "delivery-partners" do
+    resources :participants, only: %i[index]
+  end
+  get "/delivery-partners", to: redirect("/delivery-partners/participants")
```
Comment (StupidCodeFactory) [inline config/routes.rb:504]: Let's get rid of this and use the correct path in the mailer

File: db/schema.rb [hunk 5]
```diff
@@ -625,8 +634,8 @@
     t.string "training_status", default: "active", null: false
     t.string "profile_duplicity", default: "single", null: false
     t.uuid "participant_identity_id"
-    t.string "notes"
     t.string "start_term", default: "autumn_2021", null: false
+    t.string "notes"
```
Comment (StupidCodeFactory) [inline db/schema.rb:638]: ✂️

File: spec/models/delivery_partner_profile_spec.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:delivery_partner) }
+  end
+
+  describe ".create_delivery_partner_user" do
+    let(:name) { Faker::Name.name }
+    let(:email) { Faker::Internet.email }
+    let(:delivery_partner) { create(:delivery_partner) }
+    let(:created_user) { User.find_by(email: email) }
+    let(:created_delivery_partner_profile) { DeliveryPartnerProfile.find_by(user: created_user) }
```
Comment (StupidCodeFactory) [inline spec/models/delivery_partner_profile_spec.rb:20]:
```suggestion
    let(:created_delivery_partner_profile) { DeliveryPartnerProfile.find_by!(user: created_user) }
```

File: spec/models/delivery_partner_profile_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    it "creates a delivery partner user" do
+      expect {
+        DeliveryPartnerProfile.create_delivery_partner_user(name, email, delivery_partner)
+      }.to change { DeliveryPartnerProfile.count }.by(1)
+    end
+
+    it "creates a user with the correct details" do
+      DeliveryPartnerProfile.create_delivery_partner_user(name, email, delivery_partner)
+
+      expect(created_user.present?).to be(true)
```
Comment (StupidCodeFactory) [inline spec/models/delivery_partner_profile_spec.rb:31]:
```suggestion
      expect(created_user).to be_present
```

File: app/policies/delivery_partner_profile_policy.rb (hunk 2 above)
Comment (mooktakim) [reply app/policies/delivery_partner_profile_policy.rb:4]: @StupidCodeFactory it looks like there was a previous model `DeliveryPartnerProfile` which had migration and everything. Migration and model was removed, but other stuff leftover, like this policy.

**Review by StupidCodeFactory — APPROVED** [review]

**Pull request merged**